assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -996,12 +996,14 @@ public AssignmentSubmission addSubmission(String assignmentId, String submitter)
                 if (assignment.getIsGroup()) {
                     Group group = site.getGroup(submitter);
                     if (group != null && assignment.getGroups().contains(group.getReference())) {
-                        for (Member member : group.getMembers()) {
-                            AssignmentSubmissionSubmitter ass = new AssignmentSubmissionSubmitter();
-                            ass.setSubmitter(member.getUserId());
-                            submissionSubmitters.add(ass);
-                        }
-                        submission.setGroupId(submitter);
+                    	group.getMembers().stream()
+                    	.filter(m -> m.getRole().isAllowed(SECURE_ADD_ASSIGNMENT_SUBMISSION))
```

Comment:
I think this checks the permissions at a site level, and not at a group level. An instructor could have this permission disabled at site level but enabled at a group level.